livechatsnew/views.py

```diff
@@ -0,0 +1,84 @@
+import os
+from datetime import datetime
+
+from bson import ObjectId
+from django.shortcuts import redirect, reverse
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from pymongo import MongoClient
+from pydantic import ValidationError
+
+from authentication.models import CustomUser
+from .schemas import Conversation
+
+client = MongoClient(os.environ.get("MONGODB_HOST"))
+collections = client.livechats.conversations
+
+
+# Create your views here.
+@api_view(['POST'])
+def start_conversation(request, ):
+    """ Creating conversation, if it doesn`t exists, after this pk or conversation ll be used in web socket link"""
+    data = request.data
+    user_email = data.pop("email")
+    current_user = request.user
+    if request.user.email == user_email:
+        return Response({'message': 'You cannot chat with yourself'})
+
+    try:
+        participant = CustomUser.objects.get(email=user_email)
+
+    except CustomUser.DoesNotExist:
+        return Response({'message': 'You cannot chat with a non existent user'})
+
+    existing_conversation = collections.find_one({
+        "$or": [
+            {"initiator_id": current_user.id, "receiver_id": participant.id},
+            {"initiator_id": participant.id, "receiver_id": current_user.id}
+        ]
+    })
+
+    if existing_conversation:
+        return redirect(reverse('get_conversation', args=(existing_conversation["_id"],)))
```

Comment:
In an API view, it's more common to return a JSON response with the URL or ID of the resource, rather than performing an HTTP redirect. Consider returning the conversation ID and letting the client decide how to proceed